src/js/components/sidenav.js

```diff
@@ -3,125 +3,121 @@
-      this.element.querySelectorAll(this.listSelector)
-    );
-
-    menuToggles.forEach((element, index) => {
-      /**
-       * Toggle button/menu list pairings
-       * element: toggle button
-       * childMenus[index]: menu list
-       */
-
-      // Since JavaScript is available add popup semantics to toggles
-      element.setAttribute('aria-haspopup', 'true');
-
-      // Aria-expand all toggles if everything is open on init
-      if (this.openAllOnInit) {
-        element.setAttribute('aria-expanded', 'true');
-        childMenus[index].removeAttribute('hidden');
-
-        return;
-      }
-
-      // Check if this toggle been manually set to expanded in markup
-      if (element.getAttribute('aria-expanded') === 'true') {
-        // Open list matching this toggle
-        childMenus[index].removeAttribute('hidden');
-      } else {
-        element.setAttribute('aria-expanded', 'false');
-        childMenus[index].setAttribute('hidden', '');
+  static get methods() {
+    return {
+      init() {
+        console.log('Sidenav::init');
+
+        this.toggleAttribute = 'data-rvt-sidenav-toggle';
+        this.toggleSelector = `[${this.toggleAttribute}]`;
+        this.listAttribute = 'data-rvt-sidenav-list';
+        this.listSelector = `[${this.listAttribute}]`;
+        this.openAllOnInit = this.element.hasAttribute('data-rvt-sidenav-open-all');
+
+        const menuToggles = nodeListToArray(
+          this.element.querySelectorAll(this.toggleSelector)
+        );
+        const childMenus = nodeListToArray(
+          this.element.querySelectorAll(this.listSelector)
+        );
+    
+        menuToggles.forEach((element, index) => {
+          /**
+           * Toggle button/menu list pairings
+           * element: toggle button
+           * childMenus[index]: menu list
+           */
+    
+          // Since JavaScript is available add popup semantics to toggles
+          element.setAttribute('aria-haspopup', 'true');
+    
+          // Aria-expand all toggles if everything is open on init
+          if (this.openAllOnInit) {
+            element.setAttribute('aria-expanded', 'true');
+            childMenus[index].removeAttribute('hidden');
+    
+            return;
+          }
+    
+          // Check if this toggle been manually set to expanded in markup
+          if (element.getAttribute('aria-expanded') === 'true') {
+            // Open list matching this toggle
+            childMenus[index].removeAttribute('hidden');
+          } else {
+            element.setAttribute('aria-expanded', 'false');
+            childMenus[index].setAttribute('hidden', '');
+          }
+        });
+
+        // bind methods
+        Component.bindMethodToDOMElement(this, 'open', this.open);
+        Component.bindMethodToDOMElement(this, 'close', this.close);
+
+        this._handleClick = this._handleClick.bind(this);
+      },
+
+      connected() {
+        Component.dispatchComponentAddedEvent(this.element);
```

Comment:
I think I get what's going on here, but I wonder if since we can't really make a call to the parent class method because of the context of `this`. Does it make sense to still make some of these methods members of the parent `Component` class if we're just calling them from the imported object instead of `super.dispatchComponentAddedEvent()`?

I wonder if methods like this make more sense as a utility function that gets imported and then used as needed. This isn't any kind of deal-breaker for me. Just spitballing.